lib/modification/hooks/use-modifications.ts

```diff
@@ -0,0 +1,75 @@
+import {dequal} from 'dequal/lite'
+import {useEffect, useState} from 'react'
+import useSWR, {useSWRConfig} from 'swr'
+
+import {EMPTY_ARRAY} from 'lib/constants'
+import useCollection from 'lib/hooks/use-collection'
+import useUser from 'lib/hooks/use-user'
+import {getJSON, ResponseError} from 'lib/utils/safe-fetch'
+
+import type {Filter} from 'mongodb'
+
+function getURL(id: string) {
+  return `/api/db/modifications/${id}`
+}
+
+function getModifications(...ids: string[]) {
+  return Promise.all(
+    ids.map(async (id) => {
+      const res = await getJSON<CL.Modification>(getURL(id))
+      if (res.ok === true) return res.data
+      else if (res.ok === false) throw res
+    })
+  )
+}
+
+/**
+ * Read only list of modifications. First queries via the filter for all IDs that match. Then it
+ * queries the modifications individually. A list of modifications could exceed the size limit of
+ * Vercel.
```

Comment:
This documentation might be confusing to read at a later date: "read only list" sounds like an immutable list (maybe that's actually what it means?), "a list of modifications" does not unambiguously refer to the modification contents instead of the IDs.
 
```suggestion
 * Fetch a list of modification IDs for modifications that match the filter, then fetch all the individual modifications
 * separately. An array containing the full text of many modifications has been  known to exceed Vercel's 
 * response body size limit.
```